api/iteration.go

```diff
@@ -109,3 +163,30 @@ func readFileAsUTF8String(filename string) (*string, error) {
 	s := string(decodedBytes)
 	return &s, nil
 }
+
+// newIterationError executes an error message template to create a detailed
+// message for the end user. A custom error type is returned.
+func newIterationError(msgTemplate string, configured string) error {
```

Comment:
oooh. Yeah, sorry. I get it.

I think we can still use the default errorString type for this, even though we have a helper method.